gui/default/syncthing/folder/restoreVersionsModalView.html

```diff
@@ -1,5 +1,11 @@
 <modal id="restoreVersions" status="default" icon="fas fa-undo" heading="{{'Restore Versions' | translate}} ({{folderLabel(restoreVersions.folder)}})" large="yes" closeable="yes">
   <div class="modal-body">
+    <!--
+    Inform the user about loading data even before the tree is initialised
+    to avoid confusion where nothing is displayed on the screen for a while.
+    Ref: https://forum.syncthing.net/t/reversed-date-in-search-filter/21369/3
+    -->
+    <div ng-if="!restoreVersions.versions" translate>Loading data...</div>
```

Comment:
Won't this stay on screen if there are no versions to restore? Or is there something non-empty in that field then?

Reply:
It gets replaced by "There are no file versions to restore." when the versions folder is empty 🙂.

Reply:
The more I look at this code the less I understand, at least on mobile... the sizeOf() stuff makes no sense to me in the context of the new check. I'll look at a computer at some point, I don't use versioning and I'm probably missing something obvious.

Reply:
The code responsible for loading versions is convoluted, but `restoreVersions.versions` on its own only means that it runs, with no actual data whether there are any versioned files or not. That's why the whole `sizeOf()` thing is used to detect the number of versioned files. There was a lengthy discussion about this in https://github.com/syncthing/syncthing/pull/8539.